pkg/config/analysis/analyzers/authz/authorizationpolicies.go

```diff
@@ -57,10 +57,54 @@ func (a *AuthorizationPoliciesAnalyzer) Analyze(c analysis.Context) {
 	c.ForEach(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), func(r *resource.Instance) bool {
 		a.analyzeNoMatchingWorkloads(r, c, podLabelsMap)
 		a.analyzeNamespaceNotFound(r, c)
+		a.analyzeL7DenyAuthzPolicies(r, c)
 		return true
 	})
 }
 
+func (a *AuthorizationPoliciesAnalyzer) analyzeL7DenyAuthzPolicies(r *resource.Instance, c analysis.Context) {
+	ap := r.Message.(*v1beta1.AuthorizationPolicy)
+	apNs := r.Metadata.FullName.Namespace.String()
+
+	if ap.Action == v1beta1.AuthorizationPolicy_DENY {
+		for _, rule := range ap.Rules {
+			if rule.From != nil || rule.To != nil {
+				tcpRulesInFrom := false
+				tcpRulesInTo := false
+				for _, from := range rule.From {
+					if from.Source.GetPrincipals() != nil ||
+						from.Source.GetNotPrincipals() != nil || from.Source.GetIpBlocks() != nil ||
+						from.Source.GetNotIpBlocks() != nil || from.Source.GetNamespaces() != nil ||
+						from.Source.GetNotNamespaces() != nil || from.Source.GetRemoteIpBlocks() != nil ||
+						from.Source.GetNotRemoteIpBlocks() != nil {
+						tcpRulesInFrom = true
+					}
+				}
+
+				for _, to := range rule.To {
+					if to.Operation.GetPorts() != nil || to.Operation.GetNotPorts() != nil {
+						tcpRulesInTo = true
+					}
+				}
+
+				if !tcpRulesInFrom && !tcpRulesInTo {
```

Comment:
If there are no L4 rules, all the tcp traffic is blocked. IMO !L4 and only L7 are same. The policy you attached will deny all the request in ns foo (it's a correct policy). We won't get this PR's Istio Analyze message for this policy because we specifically check if there are any rules or not. We trigger the analyzer only when there is a rule in the policy which has `From` or `To`.